fedimint-api-client/src/query.rs

```diff
@@ -2,20 +2,17 @@ use std::collections::{BTreeMap, BTreeSet};
 use std::fmt::Debug;
 use std::mem;
 
-use anyhow::anyhow;
 use fedimint_core::task::{MaybeSend, MaybeSync};
 use fedimint_core::{maybe_add_send_sync, NumPeers, PeerId};
 
-use crate::api::{self, PeerError, PeerResult};
-
 /// Fedimint query strategy
 ///
 /// Due to federated security model each Fedimint client API call to the
 /// Federation might require a different way to process one or more required
 /// responses from the Federation members. This trait abstracts away the details
 /// of each specific strategy for the generic client Api code.
 pub trait QueryStrategy<IR, OR = IR> {
-    fn process(&mut self, peer_id: PeerId, response: api::PeerResult<IR>) -> QueryStep<OR>;
+    fn process(&mut self, peer_id: PeerId, response: IR) -> QueryStep<OR>;
```

Comment:
OK, so if I'm reading it right, we don't let the query strategies to see failures anymore, and instead a `retry` will just retry calls to any peer until they work.OK, so if I'm reading it right, we don't let the query strategies to see failures anymore, and instead a `retry` will just retry calls to any peer until they work.